proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+                if comp.is_solvent():
+                    return (b.flux_mass_phase_comp[t, x, p, j] == b.A_comp[t, j] * b.dens_solvent
+                            * ((prop_feed.pressure - prop_perm.pressure)
+                               - (prop_feed.pressure_osm - prop_perm.pressure_osm)))
+                elif comp.is_solute():
+                    return (b.flux_mass_phase_comp[t, x, p, j] == b.B_comp[t, j]
+                            * (prop_feed.conc_mass_phase_comp[p, j] - prop_perm.conc_mass_phase_comp[p, j]))
+
+        # ==========================================================================
+        # Final permeate mass flow rate (of solvent and solute) --> Mp,j, final = sum(Mp,j)
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.config.property_package.phase_list,
+                         self.config.property_package.component_list,
+                         doc="Permeate mass flow rates exiting unit")
+        def eq_permeate_production(b, t, p, j):
+            return (b.permeate_out[t].get_material_flow_terms(p, j)
+                    == sum(b.permeate_side[t, x].get_material_flow_terms(p, j)
+                           for x in b.feed_side.length_domain if x != 0))
+        # ==========================================================================
+        # Solute mass fraction in permeate across channel length --> Xp = Js/(Js + Jw)
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.feed_side.length_domain,
+                         self.solute_list,
+                         doc="Permeate mass fraction")
+        def eq_mass_frac_permeate(b, t, x, j):
+            if x == b.feed_side.length_domain.first():
+                return Constraint.Skip
+            else:
+                return (b.permeate_side[t, x].mass_frac_phase_comp['Liq', j]
+                        * sum(b.flux_mass_phase_comp[t, x, 'Liq', jj]
+                              for jj in b.config.property_package.component_list)
+                        == b.flux_mass_phase_comp[t, x, 'Liq', j])
+        # ==========================================================================
+        # Feed and permeate-side mass transfer connection --> Mp,j = Mf,transfer = Jj * W * L/n
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.feed_side.length_domain,
+                         self.config.property_package.phase_list,
+                         self.config.property_package.component_list,
+                         doc="Mass transfer from feed to permeate")
+        def eq_connect_mass_transfer(b, t, x, p, j):
+            if x == b.feed_side.length_domain.first():
+                return Constraint.Skip
+            else:
+                return (b.permeate_side[t, x].get_material_flow_terms(p, j)
+                        == -b.feed_side.mass_transfer_term[t, x, p, j] * b.length / nfe)
```

Comment:
Be careful here - this is only the number of finite elements. If the user selected a collocation transformation there will be more points than this.

`len(self.feed_side.length_domain)-1` might be more useful.